README.md

```diff
@@ -124,6 +124,58 @@ class Jobs::NoTransaction < QueueClassicPlus::Base
 end
 ```
 
+
+#### Callbacks
+
+Jobs support the following callbacks:
+```
+before_enqueue
+after_enqueue
+around_enqueue
+before_perform
+after_perform
+around_perform
+```
+
+- `enqueue` callbacks are called when the job is initially enqueued, but not on retries.
```

Comment:
Mainly because of implementation detail.

Retries are not implemented with the `enqueue` method, so we don't get the callback for free. https://github.com/rainforestapp/queue_classic_plus/blob/00f7487d46ba1d86048aa6787d6f4d2daef31930/lib/queue_classic_plus/base.rb#L92

Enqueuing in the future also uses a different method: https://github.com/rainforestapp/queue_classic_plus/blob/00f7487d46ba1d86048aa6787d6f4d2daef31930/lib/queue_classic_plus/base.rb#L87

So I could:
1. Refactor `queue_classic_plus` to funnel all these enqueueing methods down to a final `enqueue`, which I think would be ideal.
2. Wrap all the enqueueing methods instead of just `enqueue`. But these take different parameters, so what should we pass to the callback? We could ignore everything except `args` which are the args to the job. But would we need to differentiate in the callback between the initial enqueue and retries? 🤷 
3. Implement different callbacks for the different types of enqueues? Bleh.

I've added some caveats to the README: https://github.com/rainforestapp/queue_classic_plus/pull/60/commits/8dcdb1bcad010765adac2217c308cce65a1f6d34

These semantics aren't ideal, but since we don't have a use case yet for the `enqueue` callbacks, I decided to KIS for now.